### ISSUE TITLE
dependency fails to find lua on ubuntu 18.04

### ISSUE THREAD

**Issue by mokafolio** [mesonbuild/meson#4553]
Hi,

I have lua 5.3 installed via `apt` on ubuntu. The `dependency` function fails to find it using `dependency('lua', version: '>=5.3')` (it works on osx though for lua installed through `brew`). On Ubuntu I am only able to find it using `meson.get_compiler('cpp').find_library('lua.5.3')` which is rather clumsy. What are my options here to get it to work the same way on both osx and linux/ubuntu?

Thanks

**Comment by QuLogic** [comment]
Ubuntu has 3 versions of lua to install (presumably to exist in parallel), so they've changed the name of the pkgconfig file. You need to check for `lua-5.3`, `lua5.3`, or `lua53`, as [distributed by the `liblua5.3-dev` package](https://packages.ubuntu.com/cosmic/amd64/liblua5.3-dev/filelist). Alas, you also need to check for plain `lua` too for Fedora or homebrew which only ship one version.

**Comment by mokafolio** [comment]
So should I looke for all those 4 different variations using `dependency` and set `required` to false to then manually check if one of them succeeded? Isn't the whole point of dependency to remove that ovehead? Either way, I'll try that as a workaround for now, thank you!

**Comment by QuLogic** [comment]
Yes, ideally, but I guess no-one has had a chance to write that logic yet.

**Comment by mokafolio** [comment]
Great, just checking. I'll leave this open for now so we don't forget about it :)!

**Comment by jpakkane** [comment]
The simple way of checking what you have installed is to ask pkgconfig:

```
pkg-config --list-all|grep lua
lua53-c++                           Lua - Lua language engine
lua-5.3                             Lua - Lua language engine
lua-5.3-c++                         Lua - Lua language engine
lua5.3                              Lua - Lua language engine
lua5.3-c++                          Lua - Lua language engine
lua53                               Lua - Lua language engine
```

**Comment by mokafolio** [comment]
Okay, so what I ended up writing looks rather ugly as I am not super familiar with meson syntax yet:

`# meseon dependency function to find lua is not very consistent accross platforms yet, so
# we need to do a lot of manual lifting for now
if host_machine.system() == 'linux'
    luaDep = dependency('lua', version: '>=5.3', required: false)
    if not luaDep.found()
        luaDep = dependency('lua5.3', required: false)
        if not luaDep.found()
            luaDep = dependency('lua-5.3', required: false)
            if not luaDep.found()
                # this is our last chance, so we require this to succeed
                luaDep = dependency('lua53')
            endif
        endif
    endif
else
    #lets hope you are on a platform where this works
    luaDep = dependency('lua', version: '>=5.3')
endif`

Any improvement ideas to make it more functional and/or less ugly?

**Comment by mokafolio** [comment]
thanks, that is a great improvement!

**Comment by andy5995** [comment]
This issue happens on Ubuntu focal (20.04) and Ubuntu jammy (22.04) as well.
